### DISCUSSION TITLE
How to resolve ids from `getStructTree`?

### DISCUSSION THREAD

**Discussion by edoardocavazza** [mozilla/pdf.js#18508] [Q&A]
Hello! I am trying to create a text layer with proper HTML tags usage using the `page.getStructTree()` method, but I cannot find a way to resolve ids to actual content.

**Comment by CoderMuffin** [comment]
I think you can use the result of `page.getTextContent({ includeMarkedContent: true })`. The content in the document I had always seemed to be between between two elements marked `beginMarkedContentProps` and `endMarkedContent`). I wrote a helper function to get the content:
```ts
const textContent = await page.getTextContent({ includeMarkedContent: true }).then(x => x.items);
const findText = (id: string) => {
	let nodes = [];
	let depth = 0;
	let collecting = false;

	for (let node of textContent) {
		if ("id" in node && node.id === id) collecting = true;

		if (collecting) {
			nodes.push(node);
			
			if ("type" in node) {
				if (node.type.startsWith("begin")) depth++;
				if (node.type.startsWith("end")) depth--;

				if (depth == 0) break;
			}
		}
	}

	return nodes.map(x => ("str" in x) ? x.str : "").join("");
};
```
